pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/Common.java

```diff
@@ -501,6 +505,210 @@ private static String getRepeatedEqualsGeneration(String generatedCodeSoFar, Fie
+				} else if (f.type() == Field.FieldType.UINT32) {
+						generatedCodeSoFar +=
+       						"""
+							result = Integer.compareUnsigned($fieldName, thatObj.$fieldName);
+							if (result != 0) {
+							    return result;
+							}
+							""".replace("$fieldName", f.nameCamelFirstLower());
+
+				} else if (f.type() == Field.FieldType.FIXED64 ||
+						f.type() == Field.FieldType.INT64 ||
+						f.type() == Field.FieldType.SFIXED64 ||
+						f.type() == Field.FieldType.SINT64) {
+					generatedCodeSoFar +=
+							"""
+							result = Long.compare($fieldName, thatObj.$fieldName);
+							if (result != 0) {
+							    return result;
+							}
+							""".replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.UINT64) {
+					generatedCodeSoFar +=
+							"""
+							result = Long.compareUnsigned($fieldName, thatObj.$fieldName);
+							if (result != 0) {
+							    return result;
+							}
+							""".replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.BOOL) {
+					generatedCodeSoFar +=
+							"""
+							result = Boolean.compare($fieldName, thatObj.$fieldName);
+							if (result != 0) {
+							    return result;
+							}
+							""".replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.FLOAT) {
+					generatedCodeSoFar +=
+							"""
+							result = Float.compare($fieldName, thatObj.$fieldName);
+							if (result != 0) {
+							    return result;
+							}
+							""".replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.DOUBLE) {
+					generatedCodeSoFar +=
+							"""
+							result = Double.compare($fieldName, thatObj.$fieldName);
+							if (result != 0) {
+							     return result;
+							}
+							""".replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.STRING ||
+						f.type() == Field.FieldType.ENUM) {
+					generatedCodeSoFar += generateCompareToForObject(f);
+				} else if (f.type() == Field.FieldType.BYTES) {
+					generatedCodeSoFar +=
+							"""
+							result = $fieldName.compareTo(thatObj.$fieldName);
+							if (result != 0) {
+								return result;
+							}
+							 """.replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.MESSAGE || f.type() == Field.FieldType.ONE_OF) {
+					verifyComparable(f, destinationSrcDir);
+					generatedCodeSoFar += generateCompareToForObject(f);
+				}else {
+					throw new IllegalArgumentException("Unexpected field type for getting CompareTo - " + f.type().toString());
+				}
+			}
+		}
+		return generatedCodeSoFar.indent(DEFAULT_INDENT * 2);
+	}
+
+	@NotNull
+	private static String generateCompareToForObject(Field f) {
+		return """
+				if ($fieldName == null && thatObj.$fieldName != null) {
+				    return -1;
+				}
+				if ($fieldName != null && thatObj.$fieldName == null) {
+				    return 1;
+				}
+				if ($fieldName != null) {
```

Comment:
If I understand your suggestion correctly, this condition would ensure that the field is indeed `Comparable`. That is not necessary because the field is one of the following:
1. a primitive which maps to a comparable class (including String and Enum)
2. a wrapper for a primitive, we unwrap it and then use `compareTo` of the primitive
3. An object of a type that was verified by `Common#verifyComparable`. If the object doesn't implement `Comparable` and yet it's listed as a field that has to be included into `compareTo` of the class, there will be a build failure
See `CompareToNegativeTest#testNonComparableSubObj`

Unless I'm missing something, this explicit cast would be unnecessary

Reply:
That's why I asked the question.  It's not immediately clear if the _apparent_ type at the point this code is generated is known, by the compiler, to be `Comparable`.
Thanks for clarifying.